python/konduit/server.py

```diff
@@ -75,8 +91,32 @@ def start(self, sleep=None):
         args = self._process_extra_args(abs_path)
         process = subprocess.Popen(args=args)
         self.process = process
-        if sleep:
-            time.sleep(sleep)
+
+        # Check if the server is up or not.
+        request_timeout = 5
+        tries = int(self.start_timeout/request_timeout + 1)
+
+        started = False
+
+        for i in range(tries):
+            # This url returns status 204 with no content when the server is up.
+            try:
+                print("Checking server integrity. Tries: {} of {}".format(i + 1, tries))
+
+                r = requests.get("http://localhost:{}/healthcheck".format(self.config.serving_config.http_port),
```

Comment:
I guess right now it doesn't make much of a difference, but shall we really hard-code `localhost` here? we usually only do this in tests.

This points to a deeper issue, I guess. Where does a `Server` actually get deployed and how can we control this conveniently? Pretty sure we don't always want to do `localhost`. However, right now `Server` is just created like this, while `Client` actually requires a URL to connect. If anything, I'd assume the server determines the url and the client connects against the server according to the server's config.

This is out of scope for this PR, I think, but still something we should look into.

Reply:
@maxpumperla how about we have something like. `server.get_client()`. We don't really have any workflow right now in which we're creating the server remotely since it depends on a local `konduit.jar` file to run so I've `localhost` in there for now.

Reply:
I think the Server class would always be at localhost , no? Since we're always creating it from a local `konduit.jar` file.

Reply:
server.get_client() is essentially what I did in the other PR, but with a function, i.e. client_from_server(server). we can move that. I like this idea of yours.